runtime/src/main/java/org/corfudb/runtime/clients/BaseHandler.java

```diff
@@ -137,26 +114,196 @@ private static Object handleNotReady(CorfuMsg msg, ChannelHandlerContext ctx, IC
      * Generic handler for a server exception.
      */
     @ClientHandler(type = CorfuMsgType.ERROR_SERVER_EXCEPTION)
+    @Deprecated
     private static Object handleServerException(CorfuPayloadMsg<ExceptionMsg> msg,
-                                                ChannelHandlerContext ctx, IClientRouter r)
-            throws Throwable {
+                                                ChannelHandlerContext ctx, IClientRouter r) throws Throwable {
         log.warn("Server threw exception for request {}", msg.getRequestID(),
                 msg.getPayload().getThrowable());
         throw msg.getPayload().getThrowable();
     }
 
     /**
      * Handle a wrong cluster id exception.
+     *
      * @param msg Wrong cluster id exception message.
      * @param ctx A context the message was sent under.
-     * @param r A reference to the router.
+     * @param r   A reference to the router.
      * @return None, throw a wrong cluster id exception.
      */
     @ClientHandler(type = CorfuMsgType.WRONG_CLUSTER_ID)
+    @Deprecated
     private static Object handleWrongClusterId(CorfuPayloadMsg<WrongClusterMsg> msg,
                                                ChannelHandlerContext ctx, IClientRouter r) {
         WrongClusterMsg wrongClusterMessage = msg.getPayload();
         throw new WrongClusterException(wrongClusterMessage.getServerClusterId(),
                 wrongClusterMessage.getClientClusterId());
     }
+
+    // Protobuf region
+
+    /**
+     * Handle a ping response from the server.
+     *
+     * @param msg The ping response message.
+     * @param ctx The context the message was sent under.
+     * @param r   A reference to the router
+     * @return Always True, since the ping message was successful.
+     */
+    @ResponseHandler(type = PayloadCase.PING_RESPONSE)
+    private static Object handlePingResponse(ResponseMsg msg, ChannelHandlerContext ctx, IClientRouter r) {
```

Comment:
why not return boolean?

Reply:
As implemented, all of the client handler methods have to match the functional interface defined in `ClientResponseHandler`. Having varying return types would complicate the reflection that occurs in the `generateHandlers` method. This is very similar to what we have in `RequestHandlerMethods` for the server, except that the return type is `void`.